### ISSUE TITLE
feat: Add custom-branding to other apps besides Youtube

### ISSUE THREAD

**Issue by trmdi** [ReVanced/revanced-patches#362]
### Application

Tiktok...

### Issue

To distinguish the patched app and the original one.

### Patch

Similar to the patch for Youtube currently.

### Motivation

Many users use both patched and original apps.

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by ILoveOpenSourceApplications** [comment]
#607

**Comment by VonSparq** [comment]
I would be really interested in a patch like this as well, I think it'd be a great way to differentiate between a patched and vanilla app if the user has both.
While true that a lot of launchers have an option to change the icon/name of an app, this is not generally true of stock launchers which most people use, some because it's just what comes with their phone and it's good enough, others because since a few years using a custom launcher has been the source of some issues.
A name/icon provided by the launcher also isn't going to be respected by the rest of the OS, so finding the app in Settings can be more difficult if you have installed two of them (maybe you want notifications from the patched app but not the vanilla one, differentiating between two apps with the same icon and name isn't very intuitive, some Android skins don't even show the package name so you'd be basically guessing) and depending on the launcher/theming implementation the multitasking view and notifications might also show the actual app icon/name, not the launcher provided one.

I also think this, if possible, should be a universal patch like suggested in #2763, which alongside a patch that lets you change the apps package name (#1636) would let you install the vanilla and patched versions of all the apps and differentiate between them no matter what launcher you use.
It would even allow you to have multiple patched apps with different patches applied.
Overall I think this is a very valuable feature that would be really useful to a lot of people and have a lot of practical use cases and I would absolutely love to see it in ReVanced!

**Comment by oSumAtrIX** [comment]
> because it's just what comes with their phone and it's good enough

Well, apparently not if you want to change the icon and can't.

> isn't going to be respected by the rest of the OS

This is hardly a problem, as you'd primarily use this patch to change the appearance of your launcher.

> if you have installed two of them

You can't have the same app installed twice with the same package name

> so you'd be basically guessing

If the package name is different, there's no guessing.

Overall, this patch is a very bad idea because you'd be patching apps to change their appearance. It is possible instantly with a launcher and break updates for the app due to different signatures or entirely break the app if they have security mechanisms.

**Comment by VonSparq** [comment]
Ok, I feel like I didn't explain some things properly, sorry, let me try again!

>Well, apparently not if you want to change the icon and can't.

There are many other reasons a person might be using the stock launcher, or prefer changing the app icon in the package itself. For one, changing it from the start changes the app's appearance system wide, and not only in the launcher.
Using the stock launcher is sometimes preferable because of how tightly integrated it is into the rest of the system, for example some Android skins have a feature where your clock is displayed in the same place in the AOD, lock screen and on the home page of the launcher so the clock doesn't jump around or change fonts as you wake up and unlock the phone, this feature is broken by the usage of a custom launcher. Another feature that is often broken by custom launcher, especially on non rooted phones is the recents/multitasking view which is usually handled by the stock launcher unless the launcher provides a recents provider and the skin has a way to change it, if it does not then opening the multitasking screen takes forever on a custom launcher. There are valid reasons to stick with the stock launcher, and a custom launcher doesn't resolve this issue completely because
>This is hardly a problem, as you'd primarily use this patch to change the appearance of your launcher.

You often use it primarily, but not solely, the cases I brought up of the icon/name being wrong in the multitasking view and notifications are nearly as prominent as the launcher icon, and more so if you consider a music app that you're much more likely to interact with via the pull down menu to pause or skip a song, an email app that you'd mainly see the name and icon of as a notification or a notes, calculator or calendar app that you'd most likely use the multitasking view to switch in and out of.
>You can't have the same app installed twice with the same package name

This is true, but you absolutely can install some ReVanced apps like YouTube Music alongside it's vanilla version, and then you just have two apps that look same.
I have also mentioned that this would work great alongside the feature request #1636 which would allow this behavior for all apps, I agree that it would have been a bigger issue if #1636 was implemented without the universal custom branding patch, that's why I suggested they would work great together.
>Overall, this patch is a very bad idea because you'd be patching apps just to change their icon.

I agree! If you patch the app only to change it's icon, that would be a terrible idea! All of your explanations as to why this doesn't make sens are true and valid.
I do however feel like you are viewing this patch from a wrong angle, it obviously shouldn't be the only patch you use, #1636 is also a terrible idea in it of itself because as you mentioned in another issue, there are tools to just clone apps.

I look at it form an angle of how these two patches would work both together and alongside other patches you apply to an app. What I'm talking about here is tools that lets you patch an app with features you want, install it alongside other versions of the same app by changing it's package name and differentiate that patched app from those other versions with a system wide icon and name.
All of those features could be accomplished otherwise, you can theme apps, you can clone apps and you can patch apps with ReVanced, this would however make it much easier and more convenient to do so, all in one place!
I am absolutely not saying that anyone should use ReVanced primarily for theming or cloning, doing so is just a bad idea and there are other tools to do so, if users misuse a feature, that's on them.

**Comment by oSumAtrIX** [comment]
> You often use it primarily, but not solely

It is 99% the reason you change the icon. Minor changes such as notification icon are hardly reason to break updates or the app.

> This is true, but you absolutely can install some ReVanced apps like YouTube Music alongside it's vanilla version, and then you just have two apps that look same.

There is no reason to do that.

> also a terrible idea in it of itself because as you mentioned in another issue, there are tools to just clone apps.

That is true. There are sophisticated tools that do the job already which make such patches obsolete.

> how these two patches would work both together and alongside other patches you apply to an app

Patches are implemented without consideration of each other unless they conflict with each other. A patch, therefore, needs motivation that doesn't rely on other patches. In itself, the patch brings more drawbacks with itself than benefits. Launchers, on the other hand, do not. You'll have to weigh at the end of the day how much you want your stock launcher features or custom icons, this though, is a matter of Android and the custom launchers and what they offer in this regard, not ReVanced.

**Comment by oSumAtrIX** [comment]
@LisoUseInAIKyrios Do they cold patch apps as well or involve a better solution for that purpose? I believe shortcuts are a thing.

**Comment by VonSparq** [comment]
>Patches are implemented without consideration of each other unless they conflict with each other. A patch, therefore, needs motivation that doesn't rely on other patches. In itself, the patch brings more drawbacks with itself than benefits.

Oh, ok! This actually explains a lot!
I have found the available patches for some apps a bit strange for a while now. Some apps had features that I didn't see how would fit in with other patches, other apps seemed like they were missing patches for obvious features that'd work great with already existing patches.

Though I'm a bit disappointment by this approach as considering how patches work together would obviously make for a better patched app overall, I understand and respect your decision and the way you select and decide which patches are worth implementing.
Thank you for continuing the great work you do! <3

**Comment by oSumAtrIX** [comment]
I havent looked into this patch yet but if its simple to implement I'll just do that and call it a day, but seeing YouTube having its own file names that may not be the case.
